src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -3422,100 +3328,86 @@ namespace HVACUnitaryBypassVAV {
+                    QHeater = DataLoopNode::Node(CBVAV(CBVAVNum).HeatingCoilInletNode).MassFlowRate * CpAir *
+                              (CBVAV(CBVAVNum).CoilTempSetPoint - DataLoopNode::Node(CBVAV(CBVAVNum).HeatingCoilInletNode).Temp);
                 } else {
                     QHeater = 0.0;
                 }
                 // Added None DX heating coils calling point
-                Node(CBVAV(CBVAVNum).HeatingCoilOutletNode).TempSetPoint = CBVAV(CBVAVNum).CoilTempSetPoint;
+                DataLoopNode::Node(CBVAV(CBVAVNum).HeatingCoilOutletNode).TempSetPoint = CBVAV(CBVAVNum).CoilTempSetPoint;
                 CalcNonDXHeatingCoils(CBVAVNum, FirstHVACIteration, QHeater, CBVAV(CBVAVNum).OpMode, QHeaterActual);
             } else {
                 ShowFatalError("SimCBVAV System: Invalid Heating Coil=" + CBVAV(CBVAVNum).HeatCoilType);
             }
         }
 
-        if (CBVAV(CBVAVNum).FanPlace == DrawThru) {
+        if (CBVAV(CBVAVNum).FanPlace == DataHVACGlobals::DrawThru) {
             if (CBVAV(CBVAVNum).FanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
                 HVACFan::fanObjs[CBVAV(CBVAVNum).FanIndex]->simulate(1.0 / OnOffAirFlowRatio, _, _, _);
             } else {
                 Fans::SimulateFanComponents(CBVAV(CBVAVNum).FanName, FirstHVACIteration, CBVAV(CBVAVNum).FanIndex, FanSpeedRatio);
             }
         }
-        Node(OutletNode).MassFlowRate = (1.0 - BypassDuctFlowFraction) * Node(CBVAV(CBVAVNum).MixerInletAirNode).MassFlowRate;
-        Node(OutletNode).Temp = Node(CBVAV(CBVAVNum).SplitterOutletAirNode).Temp;
-        Node(OutletNode).HumRat = Node(CBVAV(CBVAVNum).SplitterOutletAirNode).HumRat;
-        Node(OutletNode).Quality = Node(CBVAV(CBVAVNum).SplitterOutletAirNode).Quality;
-        Node(OutletNode).Press = Node(CBVAV(CBVAVNum).SplitterOutletAirNode).Press;
-        Node(OutletNode).Enthalpy = Node(CBVAV(CBVAVNum).SplitterOutletAirNode).Enthalpy;
-        Node(OutletNode).Height = Node(CBVAV(CBVAVNum).SplitterOutletAirNode).Height;
-
-        CBVAV(CBVAVNum).BypassMassFlowRate = BypassDuctFlowFraction * Node(CBVAV(CBVAVNum).MixerInletAirNode).MassFlowRate;
+        int splitterOutNode = CBVAV(CBVAVNum).SplitterOutletAirNode;
+        DataLoopNode::Node(splitterOutNode).MassFlowRateSetPoint = DataLoopNode::Node(OutletNode).MassFlowRateSetPoint;
+        DataLoopNode::Node(OutletNode) = DataLoopNode::Node(splitterOutNode);
+        DataLoopNode::Node(OutletNode).MassFlowRate =
+            (1.0 - BypassDuctFlowFraction) * DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).MassFlowRate;
+        // report variable
+        CBVAV(CBVAVNum).BypassMassFlowRate = BypassDuctFlowFraction * DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).MassFlowRate;
+        // initialize bypass duct connected to mixer or plenum with flow rate and conditions
+        if (CBVAV(CBVAVNum).plenumIndex > 0 || CBVAV(CBVAVNum).mixerIndex > 0) {
```

Comment:
When connected to a mixer or return plenum, place the mass flow rate on the mixer/plenum node. This means the toal return air to the air loop will be the full flow, not the flow to just the zone terminal units. Before, the bypass was handled internally and mixed after the OA mixer. This was the crux of the problem Carrier had with this model. Now bypass air is mixed as the AirloopHVAC:OutsideAirSystem and air conditions entering the air loop are more representative of a real system.